crates/emath/src/ts_transform.rs

```diff
@@ -0,0 +1,121 @@
+use crate::{Pos2, Rect, Vec2};
+
+/// Linearly transforms positions via a translation, then a scaling.
+///
+/// [`TSTransform`] first translates points, then scales them with the scaling origin
+/// at `0, 0` (the top left corner)
+#[repr(C)]
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+#[cfg_attr(feature = "bytemuck", derive(bytemuck::Pod, bytemuck::Zeroable))]
+pub struct TSTransform {
+    /// Scaling applied first, scaled around (0, 0).
+    pub scaling: f32,
+
+    /// Translation amount, applied after translation.
+    pub translation: Vec2,
+}
+
+impl Eq for TSTransform {}
+
+impl Default for TSTransform {
+    #[inline]
+    fn default() -> Self {
+        Self::IDENTITY
+    }
+}
+
+impl TSTransform {
+    pub const IDENTITY: Self = Self {
+        translation: Vec2::ZERO,
+        scaling: 1.0,
+    };
+
+    /// The translation is applied first, then scaling around 0, 0.
+    #[inline]
+    pub fn new(translation: Vec2, scaling: f32) -> Self {
+        Self {
+            translation,
+            scaling,
+        }
+    }
+
+    #[inline]
+    pub fn from_translation(translation: Vec2) -> Self {
+        Self::new(translation, 1.0)
+    }
+
+    #[inline]
+    pub fn from_scaling(scaling: f32) -> Self {
+        Self::new(Vec2::ZERO, scaling)
+    }
+
+    /// Reverses the transformation from screen space to layer space.
+    ///
+    /// ```
+    /// # use emath::{pos2, vec2, TSTransform};
+    /// let p1 = pos2(2.0, 3.0);
+    /// let p2 = pos2(12.0, 5.0);
+    /// let ts = TSTransform::new(vec2(2.0, 3.0), 2.0);
+    /// let inv = ts.inverse();
+    /// assert_eq!(inv.mul_pos(p1), pos2(0.0, 0.0));
+    /// assert_eq!(inv.mul_pos(p2), pos2(5.0, 1.0));
+    /// ```
+    #[inline]
+    pub fn inverse(&self) -> Self {
+        Self::new(-self.translation / self.scaling, 1.0 / self.scaling)
+    }
+
+    /// Transforms the given coordinate by translation then scaling.
```

Comment:
We don't need to repeat the order in all docstrings. The point is more "transforms a position"